sdk/include/opentelemetry/sdk/common/empty_attributes.h

```diff
@@ -0,0 +1,21 @@
+#include "opentelemetry/trace/key_value_iterable_view.h"
+
+#include <map>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+/**
+ * Maintain a static empty map that represents empty (default) attributes.
+ * Helps to avoid constructing a new empty map everytime a call is made with default attributes.
+ */
+static const opentelemetry::trace::KeyValueIterableView<std::map<std::string, int>>
+    &GetEmptyAttributes() noexcept
+{
+  static const std::map<std::string, int>* map = new std::map<std::string, int>;
```

Comment:
Don't call `new` here, this will leak. It should with a regular `static const std::map<std::string, int> map;` Or even shorter, what you had initially:
```c++
static const nostd::shared_ptr<trace_api::KeyValueIterableView<std::map<std::string, int>>> kDefaultAttributes({});
```

Reply:
I messaged you about this on Gitter as well, but when I used my initial syntax the test would fail, and I thought it was due to the memory not being allocated properly. I think using `tatic const std::map<std::string, int> map;` will work though!

Reply:
Using a static map or nostd::shared_ptr may cause problems on destruction - the order in which destructors of static objects are called is undermined, so it is possible for static object to be destroyed while it is used if it has nontrivial destructor.

Based on Google style guide, it is preffered to leak memory for static objects (using a function local static initialzation as const static T& = *new T; so dtor for T is never called) - https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

Reply:
@pyohannes please let me know your thoughts on this!

Reply:
@IlyaKobelevskiy is right. However, you might be able to use this:
```c++
static const std::array<std::pair<std::string, int>, 0> array;                
static const opentelemetry::trace::KeyValueIterableView<std::array<std::pair<std::string, int>, 0>> kEmptyAttributes(array);
```

`std:array` has a trivial constructor. I'd like to go without leaking memory, if possible. This makes it easier to use memory checkers.

Reply:
@pyohannes, great suggestion!
If possible, I think it would also be good to define those variables as constexpr.

Reply:
I've updated the code to use `std::array` instead of `std::map`. This required changing the return type of the function as well. I also changed `array` to `constexpr`, but I wasn't able to change `kEmptyAttributes` to `constexpr`. Please let me know what you think @pyohannes @IlyaKobelevskiy

Reply:
Actually, it looks like I can't use `constexpr` for `array` either since `std::array<std::pair<std::string,int>,0>'` is not a literal type. I'll change it back to `const`.